src/datum_blocktemplates.c

```diff
@@ -331,7 +331,7 @@ T_DATUM_TEMPLATE_DATA *datum_gbt_parser(json_t *gbt) {
 
 void *datum_gateway_fallback_notifier(void *args) {
 	CURL *tcurl = NULL;
-	char userpass[512];
+	char userpass[1024];
```

Comment:
There' further inconsistency.
datum_conf.h says:
```
typedef struct {
        char bitcoind_rpcuser[256];
        char bitcoind_rpcpassword[256];
```
datum_conf.c has max_string_length set to ```128``` and it does truncate to 128

so maximum userpass would be even less: 127*2+2

Reply:
Indeed, I was mixing it up with pool_host/pubkey.

Just be careful with the one case it's reused for a file path.

Reply:
Aside from the path reuse case. How about we populate the ```userpass``` once at startup, and access as a config variable, instead of reconstructing it all the time in 5 files?

Reply:
Next commit shows what I meant. Do the userpass thing just one time, keep it global, similarly to how is set once ```datum_protocol_global_timeout_ms```. This eliminates all those multiple snprintf()s
I've also thrown in some other goodies:
- in order to stop populating userpass over and over again, populate once and reuse
- in order to eliminate discrepancies between char array sizes, use sizeof() in max_string_length of datum_config_options
- introduce a warning in case a config option is truncated
- where userpass was reused for a filepath, create a separate variable
- reduce user and pass to 128 bytes
- exit during config upon parsing error, makes more sense to me

Reply:
Is this fine?